### PR TITLE
MLDB-2100 http arbitrary slow timeout

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: plugins/fetcher.cc [hunk 1]
```diff
@@ -93,7 +93,9 @@ struct FetcherFunction: public ValueFunctionT<FetcherArgs, FetcherOutput> {
         FetcherOutput result;
         Utf8String url = args.url;
         try {
-            filter_istream stream(url.rawString(), { { "mapped", "true" } });
+                filter_istream stream(url.rawString(),
+                                      { { "mapped", "true" },
+                                        { "httpArbitraryTooSlowAbort", "1"} });
```
Comment (guyd) [inline plugins/fetcher.cc]: Why not "true".  Also, filter_istream.h has a list of the _known_ options.  You should add yours there.

File: http/http_rest_proxy.cc [hunk 2]
```diff
@@ -306,7 +308,8 @@ perform(const std::string & verb,
         bool exceptions,
         OnData onData,
         OnHeader onHeader,
-        bool followRedirect) const
+        bool followRedirect,
+        bool arbitraryTooSlowAbort) const
```
Comment (guyd) [inline http/http_rest_proxy.cc]: Since this is a bool, I think that `abortOnSlowConnection` is more appropriate.

Comment (guyd) [inline http/http_rest_proxy.cc]: Also, we should use the same name everywhere, some are prefixed with http some are not.

**Comment by guyd** [comment]
+1 beside a few minor comment.  Would it be possible to add a test as well?

File: http/http_rest_proxy.cc (hunk 2 above)
Comment (FinchPowers) [inline http/http_rest_proxy.cc]: dans le "http_rest" mettre le préfixe "http" me semble su
superflu.

Comment (guyd) [inline http/http_rest_proxy.cc]: C'est juste que ca aide a faire des greps quand on cherche comment un parametre est utilise.

Comment (jraby) [inline http/http_rest_proxy.cc]: +1 for `abortOnSlowConnection`

File: plugins/fetcher.cc (hunk 1 above)
Comment (FinchPowers) [inline plugins/fetcher.cc]: Où est cette liste?

**Comment by FinchPowers** [comment]
Je regarde pour le test.

File: plugins/fetcher.cc (hunk 1 above)
Comment (guyd) [inline plugins/fetcher.cc]: https://github.com/mldbai/mldb/blob/master/vfs/filter_streams.h#L208

File: http/http_rest_proxy.cc (hunk 2 above)
Comment (guyd) [inline http/http_rest_proxy.cc]: yup

**Comment by FinchPowers** [comment]
@guyd le test est écrit.

**Comment by guyd** [comment]
+1 pour le test

File: testing/MLDB-2100_fetcher_timeout_test.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# accept connections from outside
+try:
+    (clientsocket, address) = serversocket.accept()
+except socket.timeout:
+    mldb.log("MLDB did not contact the socket")
+    raise
+
+# now do something with the clientsocket
+# in this case, we'll pretend this is a threaded server
+ct = client_thread(clientsocket)
```
Comment (FinchPowers) [inline testing/MLDB-2100_fetcher_timeout_test.py]: Python is the server, MLDB is the client.
From the python server point of view, this is the socket to the client and the thread that manages the client.

I copied python doc without paying too much attention to the variable names and comments.

**Pull request merged**